sqlserver/tests/test_activity.py

```diff
@@ -0,0 +1,252 @@
+import json
+import os
+import time
+from concurrent.futures.thread import ThreadPoolExecutor
+from copy import copy
+
+import mock
+import pytest
+
+from datadog_checks.base.utils.db.utils import DBMAsyncJob, default_json_event_encoding
+from datadog_checks.sqlserver import SQLServer
+
+from .common import CHECK_NAME
+from .utils import not_windows_ci, windows_ci
+
+try:
+    import pyodbc
+except ImportError:
+    pyodbc = None
+
+ACTIVITY_JSON_PLANS_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "activity")
+
+
+@pytest.fixture(autouse=True)
+def stop_orphaned_threads():
+    # make sure we shut down any orphaned threads and create a new Executor for each test
+    DBMAsyncJob.executor.shutdown(wait=True)
+    DBMAsyncJob.executor = ThreadPoolExecutor()
+
+
+@pytest.fixture
+def dbm_instance(instance_docker):
+    instance_docker['dbm'] = True
+    # set a very small collection interval so the tests go fast
+    instance_docker['query_activity'] = {'enabled': True, 'run_sync': True, 'collection_interval': 0.1}
+    # do not need query_metrics for these tests
+    instance_docker['query_metrics'] = {'enabled': False}
+    return copy(instance_docker)
+
+
+@pytest.fixture
+def instance_sql_msoledb_dbm(instance_sql_msoledb):
+    instance_sql_msoledb['dbm'] = True
+    instance_sql_msoledb['min_collection_interval'] = 1
+    instance_sql_msoledb['query_activity'] = {'enabled': True, 'run_sync': True, 'collection_interval': 0.1}
+    # not needed for this test
+    instance_sql_msoledb['query_metrics'] = {'enabled': False}
+    instance_sql_msoledb['tags'] = ['optional:tag1']
+    return instance_sql_msoledb
+
+
+@not_windows_ci
+@pytest.mark.integration
+@pytest.mark.usefixtures('dd_environment')
+def test_collect_activity(aggregator, instance_docker, dd_run_check, dbm_instance):
+    _run_test_collect_activity(aggregator, instance_docker, dd_run_check, dbm_instance)
+
+
+@windows_ci
+@pytest.mark.integration
+def test_collect_activity_windows(aggregator, instance_docker, dd_run_check, instance_sql_msoledb_dbm):
+    _run_test_collect_activity(aggregator, instance_docker, dd_run_check, instance_sql_msoledb_dbm)
+
+
+def _run_test_collect_activity(aggregator, instance_docker, dd_run_check, dbm_instance):
+    check = SQLServer(CHECK_NAME, {}, [dbm_instance])
+    query = "select * from things"
+    bob_conn = _get_conn_for_user(instance_docker, "bob")
+    with bob_conn.cursor() as cursor:
+        cursor.execute("USE {}".format("datadog_test"))
+        cursor.execute(query)
+
+    fred_conn = _get_conn_for_user(instance_docker, "fred")
+    with fred_conn.cursor() as cursor:
+        cursor.execute("USE {}".format("datadog_test"))
+        cursor.execute(query)
+
+    time.sleep(2)  # wait for coll interval
```

Comment:
If we're running synchronously (given `'run_sync': True`) do we need to wait for the collection interval?

Reply:
When we look for old open transactions that is based on time https://github.com/DataDog/integrations-core/pull/10610/files#diff-081417075be9ae91f8c736c79e42f51df5eb4aeb68258a1c0be7fbf4290883faR133